src/ImageSharp/Image.LoadPixelData.cs

```diff
@@ -103,13 +98,9 @@ private static Image<TPixel> LoadPixelData<TPixel>(Configuration config, Span<by
         public static Image<TPixel> LoadPixelData<TPixel>(Configuration config, TPixel[] data, int width, int height)
             where TPixel : struct, IPixel<TPixel>
         {
-            int count = width * height;
-            Guard.MustBeGreaterThanOrEqualTo(data.Length, count, nameof(data));
-
-            var image = new Image<TPixel>(config, width, height);
-            SpanHelper.Copy(data, image.GetPixelSpan(), count);
-
-            return image;
+            // There's an implict cast to Span from Array
+            // Should we remove this overload and expose Span<TPixel> ?
+            return LoadPixelData(config, new Span<TPixel>(data), width, height);
```

Comment:
See #489 also

Reply:
The implicit cast doesn't seem to pick up in this case. Removing this comment (should revisit later).